docs/programming-oak.md

```diff
@@ -132,11 +135,12 @@ fn main() {
 ```
 <!-- prettier-ignore-end -->
 
-The autogenerated code includes two key chunks.
+The autogenerated code includes three distinct sections.
```

Comment:
It's unclear (to me, at least) what "section" actually means here (granted, "chunks" was equally vague). Perhaps we should say that it generates three objects: a trait, a struct, and another struct, and make this a bit more structured (bullet list, or sub-headings)?

Reply:
Shifted to "parts" and added a bullet list summary at the start.